src/downshift.js

```diff
@@ -561,18 +533,50 @@ class Downshift extends Component {
   keyDownHandlers = {
     ArrowDown(event) {
       event.preventDefault()
-      const amount = event.shiftKey ? 5 : 1
-      this.moveHighlightedIndex(amount, {
-        type: stateChangeTypes.keyDownArrowDown,
-      })
+
+      if (this.getState().isOpen) {
+        const amount = event.shiftKey ? 5 : 1
+        this.moveHighlightedIndex(amount, {
+          type: stateChangeTypes.keyDownArrowDown,
+        })
+      } else {
+        const self = this
+        this.toggleMenu({type: stateChangeTypes.keyDownArrowDown}, () => {
+          const itemCount = self.getItemCount()
+          if (itemCount > 0) {
+            const {highlightedIndex} = self.getState()
+            const newIndex =
+              typeof highlightedIndex === 'number'
+                ? getNextWrappingIndex(1, highlightedIndex, itemCount)
+                : 0
+            self.setHighlightedIndex(newIndex)
+          }
+        })
+      }
     },
 
     ArrowUp(event) {
       event.preventDefault()
-      const amount = event.shiftKey ? -5 : -1
-      this.moveHighlightedIndex(amount, {
-        type: stateChangeTypes.keyDownArrowUp,
-      })
+
+      if (this.getState().isOpen) {
```

Comment:
The only difference I see between these handler and ArrowDown is the `stateChangeType` and `newIndex`. Does it make sense to move to a method and pass the correct newIndex value?

Maybe it complicates things as `newIndex` is calculated from state after the menu gets toggled. And repeating something twice could be better than a wrong abstraction.